block_access.cpp

```diff
@@ -944,7 +944,9 @@ int deleteRelCatEntry(recId relcat_recid, Attribute relcat_rec[6]) {
 	relcat_rec[2].nval = relcat_rec[2].nval - 1;
 	setRecord(relcat_rec, 4, 0);
 
-	int no_of_attrs = relcat_rec[1].nval;
+	getRecord(relcat_rec, 4, relcat_recid.slot);
+	int no_of_attrs = relcat_rec[2].nval;
```

Comment:
The purpose of deleteRelCatEntry, as I see it, is setting the slot of relation's relCatEntry as 0 and updating the number of records in relation catalog's relCatEntry's number of records and attribute Catalog's relCatEntry's number of records. When we need to delete a relation catalog entry of a deleting relation, the attribute catalog's relCatEntry in the relation catalog block must also be modified. And the decrease in it's no. of records is ***not always*** the ***no. of attrs of relation catalog*** because the relation we are dropping is ***not always relation catalog***.

Reply:
And the `no_of_attrs` we need must be obtained from the no of attrs of the relation we delete. Which can be any number (2, 3, 4). whereas the no. of attributes of the relation catalog is always `6.`

Reply:
I need to add one more change to this as well. I am setting the slot as unoccupied after getrecord is done. If not, the expected results are not coming. Please allow me to do this @clifordjoshy.